cmd/pod-scaler/consumer.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/interrupts"
+)
+
+func newReloader(name string, cache cache) *cacheReloader {
+	reloader := &cacheReloader{
+		name:  name,
+		cache: cache,
+		logger: logrus.WithFields(logrus.Fields{
+			"component": "reloader",
+			"metric":    name,
+		}),
+		lock: &sync.RWMutex{},
+	}
+	interrupts.TickLiteral(reloader.reload, 10*time.Minute)
+	return reloader
+}
+
+type cacheReloader struct {
+	name   string
+	cache  cache
+	logger *logrus.Entry
+
+	lock        *sync.RWMutex
+	lastUpdated time.Time
+	lastLoaded  *CachedQuery
+	subscribers []chan<- interface{}
+}
+
+func (c *cacheReloader) subscribe(out chan<- interface{}) {
+	c.lock.Lock()
+	c.subscribers = append(c.subscribers, out)
+	c.lock.Unlock()
+}
+
+func (c *cacheReloader) reload() {
+	// technically this can race as we read the attribute and data from the handle at
+	// different times, but there doesn't seem to be an atomic call to GCS for that anyway
+	lastUpdated, err := lastUpdated(c.cache, c.name)
+	if err != nil {
+		c.logger.WithError(err).Warn("Failed to query for last cache update time, won't reload this tick.")
+		return
+	}
+	c.lock.RLock()
+	lastSeen := c.lastUpdated
```

Comment:
That doesn't seem to be a read

Reply:
What do you mean? We read `c.lastUpdated`